File: easy/121-Best_Time_to_Buy_and_Sell_Stock.py
def max_profit(self, prices: List[int]) -> int:
    low = prices[0]
    max_profit = 0
    
    for value in prices:
        if value < low:
            low = value
        elif max_profit < (value - low):
            max_profit = (value - low)
    
    return max_profit

File: easy/121-Best_Time_to_Buy_and_Sell_Stock.java
public int maxProfit(int[] prices) {
	int maxProfit = 0;
	int low = prices[0];
	
	for(int a = 1; a < prices.length; a++) {
		if (prices[a] < low)
			low = prices[a];
		else if (prices[a] - low > maxProfit)
			maxProfit = prices[a] - low;
	}
	
	return maxProfit;
}